azurecompute/src/main/java/org/jclouds/azurecompute/compute/AzureComputeServiceAdapter.java

```diff
@@ -107,6 +107,9 @@
 
       final String storageAccountName = templateOptions.getStorageAccountName().get();
       final String virtualNetworkName = templateOptions.getVirtualNetworkName().get();
+      final String reservedIPAddress = templateOptions.getReservedIPName().isPresent()
+              ? templateOptions.getReservedIPName().get()
+              : null;
```

Comment:
Change to `final String reservedIPAddress = templateOptions.getReservedIPName().orNull();`